### ISSUE TITLE
When using CSP in a Vue3 Vite build it will block

### ISSUE THREAD

**Issue by gerardvde** [intlify/vue-i18n-next#1457]
### Reporting a bug?

When using CSP in a Vue3 vite build this error is thrown:
EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self' 'nonce-FMO9QrSTmJ76tL87'

The "new Function" will cause this

### Expected behavior

When I build a Vue3 project with vite and i18n (version 9.2.2) and CSP Headers I will expect no blocking

### Reproduction

Repo is VPN protected

### System Info

```shell
System:                                                    
    OS: Windows 10 10.0.19045                                
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12800H       
    Memory: 11.18 GB / 31.67 GB                              
  Binaries:                                                  
    Node: 18.16.0 - C:\Program Files\nodejs\node.EXE         
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD           
    npm: 9.5.1 - C:\Program Files\nodejs\npm.CMD             
  Browsers:                                                  
    Edge: Spartan (44.19041.1266.0), Chromium (114.0.1823.67)
    Internet Explorer: 11.0.19041.1566                       
  npmPackages:                                               
    @vitejs/plugin-vue: ^4.2.3 => 4.2.3                      
    @vue/eslint-config-prettier: ^7.1.0 => 7.1.0             
    @vue/eslint-config-typescript: ^11.0.3 => 11.0.3         
    @vue/test-utils: ^2.2.7 => 2.4.0                         
    @vue/tsconfig: ^0.4.0 => 0.4.0                           
    vite: ^4.3.9 => 4.3.9
    vitest: ^0.32.0 => 0.32.2
    vue: ^3.3.4 => 3.3.4
    vue-axios: ^3.5.2 => 3.5.2
    vue-i18n: ^9.2.2 => 9.2.2
    vue-router: ^4.2.2 => 4.2.2
    vue-tsc: ^1.6.5 => 1.8.3
```

### Screenshot

![image](https://github.com/intlify/vue-i18n-next/assets/5820791/32a7d756-c1a2-498a-aad4-bdd8c3e7492b)

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n-next/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussions](https://github.com/intlify/vue-i18n-next/discussions)

**Comment by kazupon** [comment]
CSP issue will be resolved in v9.3 with JIT compilation.
Currently v9.3 beta has been released, so you can try the latest beta version.
https://vue-i18n.intlify.dev/guide/advanced/optimization.html

If you try it, please also use the latest version of unplugin-vue-i18n.
https://github.com/intlify/bundle-tools/tree/main/packages/unplugin-vue-i18n#jitcompilation

**Comment by nicooprat** [comment]
I can confirm there's no more `new Function` when doing the following:

* upgrade to latest beta: `pnpm i -S vue-i18n@v9.3.0-beta.24`
* add `define: { __INTLIFY_JIT_COMPILATION__: true }` in `vite.config.ts`

👌
